cmd/agent/subcommands/streamlogs/command.go

```diff
@@ -143,11 +163,34 @@ func streamRequest(url string, body []byte, onChunk func([]byte)) error {
 	return e
 }
 
+// openFileForWriting opens a file for writing
 func openFileForWriting(filePath string) (*os.File, *bufio.Writer, error) {
 	f, err := os.OpenFile(filePath, os.O_APPEND|os.O_CREATE|os.O_WRONLY, 0644)
 	if err != nil {
 		return nil, nil, fmt.Errorf("error opening file %s: %v", filePath, err)
 	}
-	bufWriter := bufio.NewWriter(f)
+	bufWriter := bufio.NewWriter(f) // default 4096 bytes buffer
 	return f, bufWriter, nil
 }
+
+// doPostChunkedWithTimeout is a wrapper around requests that stream chunked data with a timeout
+func doPostChunkedWithTimeout(ctx context.Context, c *http.Client, url string, contentType string, body io.Reader, duration time.Duration, onChunk func([]byte)) error {
+	var cancel context.CancelFunc
+	ctx, cancel = context.WithTimeout(ctx, duration)
+	defer cancel()
+
+	done := make(chan bool)
+	var err error
+
+	go func() {
+		err = util.DoPostChunked(c, url, contentType, body, onChunk)
+		done <- true
+	}()
+
+	select {
+	case <-ctx.Done():
```

Comment:
Instead of relying on a context to try to cancel the read, would it make more sense to rely on go's HTTP client built in timeout? [We create the client here](https://github.com/DataDog/datadog-agent/blob/b386591b15faab716428d43633f83324ca0ff610/pkg/api/util/doget.go#L46) and you can set a timeout [like this](https://stackoverflow.com/a/25344458/1152375)